pkg/engine/renderer/enhancer_test.go

```diff
@@ -0,0 +1,277 @@
+		assert.Equal(t, string(meta.PlanUID), sfs.Annotations[kudo.PlanUIDAnnotation])
+		assert.Equal(t, string(meta.PlanUID), sfs.Spec.Template.Annotations[kudo.PlanUIDAnnotation])
+
+		// Verify that annotations are added
+		assert.Equal(t, "kudo", sfs.Labels[kudo.HeritageLabel])
+		assert.Equal(t, "kudo", sfs.Spec.Template.Labels[kudo.HeritageLabel])
+		assert.Equal(t, "kudo", sfs.Spec.VolumeClaimTemplates[0].Labels[kudo.HeritageLabel])
+		assert.Equal(t, "kudo", sfs.Spec.VolumeClaimTemplates[1].Labels[kudo.HeritageLabel])
+
+		// Verify that existing labels are not removed
+		assert.Equal(t, "app-type", sfs.Spec.Template.Labels["app"])
+		assert.Equal(t, "vct1label", sfs.Spec.VolumeClaimTemplates[0].Labels["vct1"])
+		assert.Equal(t, "vct2label", sfs.Spec.VolumeClaimTemplates[1].Labels["vct2"])
+	}
+}
+
+func TestEnhancerApply_embeddedMetadataCronjob(t *testing.T) {
+
+	tpls := map[string]string{
+		"cron": resourceAsString(cronjob("cronjob", "default")),
+	}
+
+	meta := metadata()
+
+	e := &DefaultEnhancer{
+		Scheme: utils.Scheme(),
+	}
+
+	objs, err := e.Apply(tpls, meta)
+	if err != nil {
+		t.Errorf("failed to apply template %s", err)
+	}
+
+	for _, o := range objs {
+		unstructMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(o)
+		if err != nil {
+			t.Errorf("failed to parse object to unstructured: %s", err)
+		}
+		cron := &v1beta1.CronJob{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructMap, cron); err != nil {
+			t.Errorf("failed to parse unstructured to CronJob: %s", err)
+		}
+
+		// Verify that labels are added directly and on template
+		assert.Equal(t, meta.InstanceNamespace, cron.GetNamespace())
+		assert.Equal(t, string(meta.PlanUID), cron.Annotations[kudo.PlanUIDAnnotation])
+		assert.Equal(t, "kudo", cron.Labels[kudo.HeritageLabel])
+
+		assert.Equal(t, string(meta.PlanUID), cron.Spec.JobTemplate.Spec.Template.Annotations[kudo.PlanUIDAnnotation])
+		assert.Equal(t, "kudo", cron.Spec.JobTemplate.Spec.Template.Labels[kudo.HeritageLabel])
+
+		// Verify that existing labels are not removed
+		assert.Equal(t, "labelvalue", cron.Spec.JobTemplate.Labels["additional"])
+		assert.Equal(t, "app-type", cron.Spec.JobTemplate.Spec.Template.Labels["app"])
+	}
+}
+
+func TestEnhancerApply_noAdditionalMetadata(t *testing.T) {
+
+	tpls := map[string]string{
+		"pod": resourceAsString(pod("pod", "default")),
+	}
+
+	meta := metadata()
+
+	e := &DefaultEnhancer{
+		Scheme: utils.Scheme(),
+	}
+
+	objs, err := e.Apply(tpls, meta)
+	if err != nil {
+		t.Errorf("failed to apply template %s", err)
+	}
+
+	for _, o := range objs {
+		unstructMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(o)
+		if err != nil {
+			t.Errorf("failed to parse object to unstructured: %s", err)
+		}
+
+		f, ok, _ := unstructured.NestedFieldNoCopy(unstructMap, "spec", "template")
+
+		assert.Nil(t, f)
+		assert.False(t, ok, "Pod struct contains template field")
+
+		t.Logf("Tree: %v", unstructMap)
```

Comment:
this output is strange... is it adding value?   why here and not in other tests?

Reply:
Yeah, that shouldn't be there, was a left over debug output.